File: 141-linked-list-cycle/linked-list-cycle.java
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    public boolean hasCycle(ListNode head) {
        ListNode a1=head;
        ListNode a2=head;
        while(a1!=null &&a1.next!=null){
            a1=a1.next.next;
            a2=a2.next;
            if(a1==a2){
                return true;
            }
        }
        return false;
        
    }
}

File: 141-linked-list-cycle/linked-list-cycle.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    bool hasCycle(ListNode *head) {
        ListNode* A1=head;
        ListNode* A2=head;
        bool ans=false;
        while(A1!=NULL && A1->next!=NULL){
            A1=A1->next->next;
            A2=A2->next;
            if(A1==A2){
                return true;
            }
            // else{
            //     ans=false;
            // }
        }
        return false;
        
    }
};